src/java/org/apache/cassandra/service/accord/txn/TxnReference.java

```diff
@@ -43,18 +52,22 @@
 import org.apache.cassandra.service.accord.serializers.TableMetadatas;
 import org.apache.cassandra.service.accord.serializers.Version;
 import org.apache.cassandra.utils.ByteBufferUtil;
+import org.apache.cassandra.utils.ObjectSizes;
 
 import static org.apache.cassandra.db.marshal.CollectionType.Kind.SET;
 import static org.apache.cassandra.service.accord.AccordSerializers.columnMetadataSerializer;
 
 public class TxnReference
 {
     private final int tuple;
+    @Nullable
     private final TableMetadata table;
+    @Nullable
     private final ColumnMetadata column;
+    @Nullable
```

Comment:
I guess these are all technically nullable (for tests or w/ CAS requests?), but annotating them as such creates some noise for static analysis tools on this class. I wonder how hard it would be to make `TxnReference` and interface and separate the behavior of the normal case from the CAS case....